### PR TITLE
Channel autocomplete

### LINKED ISSUE #803: Channel Suggestions Only Suggesting Joined Channels
While I was working on the `/join` command I noticed that the channel suggestions you get when you type `#` are only channels which you have joined. This is rather limiting when you're using the `/join` command and want a list of the channels which you can join that match what you're typing.

### PR DESCRIPTION
**Pull request by gmsecrieru**
Should close #803

Channel autocomplete now displays all channels instead of only displaying channels which the user is already a member.

### CODE REVIEW

File: client/views/app/messagePopupConfig.coffee [hunk 1]
```diff
@@ -1,31 +1,36 @@
+@onlineUsers = new Mongo.Collection 'online-users'
```
Comment (rodrigok) [inline client/views/app/messagePopupConfig.coffee]: Why did you add this collection again?

File: server/publications/channelAutocomplete.coffee [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+	options =
+		fields:
+			_id: 1
+			name: 1
+		limit: 5
+
+	pub = this
+
+	cursorHandle = RocketChat.models.Rooms.find(query, options).observeChanges
```
Comment (rodrigok) [inline server/publications/channelAutocomplete.coffee]: Cab you change here to use a named method?

**Comment by gmsecrieru** [comment]
@rodrigok I think we are good to go now!
